### PR TITLE
feat:  add service provider entity + extension of idLookup

### PR DESCRIPTION
**Pull request by peinwag**
Implements:
- handling a of the propertyManager entity
- extension of the idLookup endpoint (new resource `propertyManager`+ possibility to consider a parent in the lookup context)

depends on https://github.com/allthings/php-prod/pull/940

### CODE REVIEW

**Comment by adieuadieu** [comment]
Just a quick thing: didn't we rename "property manager" to "service provider" in our terminology? If yes, I think we should use the latter in our method naming on the SDK.

**Comment by peinwag** [comment]
> Just a quick thing: didn't we rename "property manager" to "service provider" in our terminology? If yes, I think we should use the latter in our method naming on the SDK.

@adieuadieu right will do so - but what about the existing endpoints in the api? e.g. `/v1/property-managers` would it make sense to create aliases for them in order to be consistent with the method names in the sdk?

**Comment by adieuadieu** [comment]
> @adieuadieu right will do so - but what about the existing endpoints in the api? e.g. /v1/property-managers would it make sense to create aliases for them in order to be consistent with the method names in the sdk?

If it's quick/easy i think it makes sense?

File: src/rest/methods/serviceProvider.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    street: 'street',
+  },
+  email: 'foo@bar.de',
+  name: 'Foobar Property-manager',
+  phoneNumber: '+493434343343',
+  type: EnumServiceProviderType.craftsPeople,
+}
+
+describe('serviceProviderCreate()', () => {
+  it('should be able to create a new property-manager', async () => {
```
Comment (adashst) [inline src/rest/methods/serviceProvider.test.ts]: maybe 
```suggestion
  it('should be able to create a new service provider', async () => {
```
?

File: src/rest/methods/serviceProvider.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const data = { ...testData, externalId: generateId() }
+    const result = await client.serviceProviderCreate(data)
+
+    expect(result.name).toEqual(data.name)
+    expect(result.externalId).toEqual(data.externalId)
+  })
+})
+
+describe('serviceProviderGetById()', () => {
+  it('should be able to get a property by ID', async () => {
```
Comment (adashst) [inline src/rest/methods/serviceProvider.test.ts]:
```suggestion
  it('should be able to get a service provider by ID', async () => {
```

File: src/rest/methods/serviceProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    const { id } = await client.serviceProviderCreate(data)
+    const result = await client.serviceProviderGetById(id)
+
+    expect(result.name).toEqual(data.name)
+    expect(result.externalId).toEqual(data.externalId)
+  })
+})
+
+describe('propertyUpdateById()', () => {
+  it('should be able to update a property by ID', async () => {
```
Comment (adashst) [inline src/rest/methods/serviceProvider.test.ts]:
```suggestion
  it('should be able to update a service provider by ID', async () => {
```

**Pull request merged**